### PR TITLE
feat: Encryption tests for integration-browser

### PR DESCRIPTION
**Pull request by seebees**
https://github.com/awslabs/aws-crypto-tools-test-vector-framework defines an encryption manifest.
This manifest should be used to maintain cross compatibility of language implementations.

Then integrate this into the CI integration tests.
Also update package.json to better organize parameters.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: modules/integration-browser/src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  writeFileSync(`${fixtures}/decrypt_tests.json`, JSON.stringify([]))
+  writeFileSync(`${fixtures}/encrypt_tests.json`, JSON.stringify([]))
+  writeFileSync(`${fixtures}/decrypt_oracle.json`, JSON.stringify(decryptOracle))
+
+  if (command === 'decrypt') {
+    const { vectorFile } = argv
+    // @ts-ignore
+    await buildDecryptFixtures(fixtures, vectorFile, testName, slice)
+  } else if (command === 'encrypt') {
```
Comment (johnwalker) [inline modules/integration-browser/src/cli.ts]: Why ignore the await?

Similarly, I noticed a few eslint disables through the tests, for unused variables. Can we keep these linters and checks on?

File: modules/integration-browser/src/integration.encrypt.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-env jasmine */
```
Comment (johnwalker) [inline modules/integration-browser/src/integration.encrypt.test.ts]: Is there any way to apply this to a directory?

File: modules/integration-browser/src/types.ts [hunk 3]
```diff
@@ -13,11 +13,21 @@
  * limitations under the License.
  */
 
-export interface ManifestList {
-  manifest: Manifest
+import { EncryptionContext } from '@aws-crypto/client-browser' // eslint-disable-line no-unused-vars
```
Comment (johnwalker) [inline modules/integration-browser/src/types.ts]: It looks like it's being used. Would you remove the eslint disable?

File: modules/integration-browser/src/integration.encrypt.test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            'Content-Type': 'application/octet-stream',
+            'Accept': 'application/octet-stream'
+          },
+          body: cipherMessage
+        })
+        const body = await response.arrayBuffer()
+        needs(response.ok, `Failed to decrypt: ${toUtf8(body)}`)
+        expect(plainText).toEqual(new Uint8Array(body))
+      } catch (e) {
+        if (!notSupportedMessages.includes(e.message)) throw e
```
Comment (ragona) [inline modules/integration-browser/src/integration.encrypt.test.ts]: I don't suppose there's a better way to handle specific error types that isn't just string parsing? It looks like if one of those error messages changes the logic here breaks.

File: modules/integration-browser/src/build_encrypt_fixtures.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import { randomBytes } from 'crypto'
+import {
+  AlgorithmSuiteIdentifier, // eslint-disable-line no-unused-vars
+  EncryptionContext // eslint-disable-line no-unused-vars
+} from '@aws-crypto/client-browser'
+import { URL } from 'url'
+import { readFileSync, writeFileSync } from 'fs'
+import got from 'got'
+
+export async function buildEncryptFixtures (fixtures: string, manifestFile: string, keyFile: string, testName: string, slice: string) {
```
Comment (johnwalker) [inline modules/integration-browser/src/build_encrypt_fixtures.ts]: Would you add some comments to this? I need help understanding the choice of start, end, 10, the randomBytes, and what the goal of the method is.

File: modules/integration-browser/src/build_encrypt_fixtures.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  const [start = 0, end = 9999] = (slice || '').split(':').map(n => parseInt(n, 10))
```
Comment (ragona) [inline modules/integration-browser/src/build_encrypt_fixtures.ts]: Nitpick: This smells funny.

File: modules/integration-browser/src/integration.encrypt.test.ts (hunk 2 above)
Comment (seebees) [reply modules/integration-browser/src/integration.encrypt.test.ts]: Not that I know of

File: modules/integration-browser/src/types.ts (hunk 3 above)
Comment (seebees) [reply modules/integration-browser/src/types.ts]: It is technically not being used.  From the JavaScript side, this is not used.  There should be a better solution to this eventually, but not at this time.

File: modules/integration-browser/src/integration.encrypt.test.ts (hunk 4 above)
Comment (seebees) [reply modules/integration-browser/src/integration.encrypt.test.ts]: There maybe.  But that would be an extensive change to the ESDK that I may make...

File: modules/integration-browser/src/build_encrypt_fixtures.ts (hunk 6 above)
Comment (seebees) [reply modules/integration-browser/src/build_encrypt_fixtures.ts]: Very funny.  Just me working around some cmd line parsing :(
Humans are difficult interfaces...

File: modules/integration-browser/src/cli.ts (hunk 1 above)
Comment (seebees) [reply modules/integration-browser/src/cli.ts]: It is not being used from a JS perspective, only Typescript type...

The function does not return anything, so there is nothing to "get"

File: modules/integration-browser/src/build_encrypt_fixtures.ts (hunk 5 above)
Comment (seebees) [reply modules/integration-browser/src/build_encrypt_fixtures.ts]: added

**Pull request merged**